Bottom navigation bar list image

dhruva-suryadigital opened the pull request.

dhruva-suryadigital committed:
Implement Bottom navigation bar

dhruva-suryadigital committed:
Add scrollable list and image

Implement scrollable list and a button for HomePage
Add circular image for ProfilePage

Amulya-Joshi commented on lib/color.dart:
```diff
@@ -0,0 +1,5 @@
+import 'package:flutter/material.dart';
+
+Color? primaryColor(int shade) {
```
Why is this nullable?

dhruva-suryadigital replied:
The `[]` returns a Color value from a map and if value doesn't exist, it returns null.

Amulya-Joshi commented on lib/main.dart:
```diff
@@ -1,115 +1,207 @@
+//import 'package:flutter/cupertino.dart';
```
@dhruva-suryadigital Never leave unnecessary commented out code on your PR. Please review your PR before requesting the reviewer.

Amulya-Joshi commented on lib/main.dart:
```diff
@@ -1,115 +1,207 @@
+          )
+        ]),
+    const Text('Search Page',
+        style: TextStyle(fontSize: 35, fontWeight: FontWeight.bold)),
```
Please format with `,`.

Amulya-Joshi commented on lib/main.dart:
```diff
@@ -1,115 +1,207 @@
     });
   }
 
+  List<Widget> pages = <Widget>[
```
```suggestion
  List<Widget> _pages = <Widget>[
```

Amulya-Joshi commented on lib/main.dart:
```diff
@@ -1,115 +1,207 @@
 
+  List<Widget> pages = <Widget>[
+    Column(
+        mainAxisAlignment: MainAxisAlignment.start,
```
Is this necessary? Please check again.

Amulya-Joshi commented on lib/main.dart:
```diff
@@ -1,115 +1,207 @@
+                Container(
+                  decoration: BoxDecoration(
+                      color: colors.primaryColor(100),
+                      borderRadius: BorderRadius.circular(10)),
```
Format code.

Amulya-Joshi commented on lib/main.dart:
```diff
@@ -1,115 +1,207 @@
+                    child: Text("Entry 2"),
+                  ),
+                ),
+                Container(
```
Please create a custom widget for this.

Amulya-Joshi commented on lib/main.dart:
```diff
@@ -1,115 +1,207 @@
+          decoration: BoxDecoration(
+              image: const DecorationImage(
+                image: AssetImage('assets/images/abstract.jpg'),
+// image: NetworkImage(
```
Please do not leave unnecessary comments on the code.

Amulya-Joshi commented on lib/main.dart:
```diff
@@ -1,115 +1,207 @@
+                image: AssetImage('assets/images/abstract.jpg'),
+// image: NetworkImage(
+//     'https://images.unsplash.com/photo-1618005182384-a83a8bd57fbe?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=2864&q=80'),
+                fit: BoxFit.cover,
```
What does Boxfit.cover do?

Amulya-Joshi commented on lib/main.dart:
```diff
@@ -1,115 +1,207 @@
+                fit: BoxFit.cover,
+              ),
+              color: colors.primaryColor(900),
+              //shape: BoxShape.circle,
```
This should not be here.

Amulya-Joshi commented on lib/main.dart:
```diff
@@ -1,115 +1,207 @@
     );
   }
+
+//   Center containerImage() {
```
Please remove this.

Amulya-Joshi commented on lib/main.dart:
```diff
@@ -1,115 +1,207 @@
+              label: "search",
             ),
+            BottomNavigationBarItem(
+                icon: Icon(Icons.shopping_cart), label: "cart"),
```
Code not formatted.

Amulya-Joshi commented on lib/main.dart:
```diff
@@ -1,115 +1,207 @@
+                icon: Icon(Icons.shopping_cart), label: "cart"),
+            BottomNavigationBarItem(icon: Icon(Icons.person), label: "account"),
           ],
+          type: BottomNavigationBarType.fixed,
```
What does this do?

dhruva-suryadigital replied:
The `BottomNavigationBarItem`s  will have fixed width and labels will be visible.

Amulya-Joshi commented on pubspec.yaml:
```diff
@@ -59,9 +59,9 @@ flutter:
-  #   - images/a_dot_ham.jpeg
+  assets:
+    - assets/
+    - assets/images/
```
Check if both the lines is necessary.

dhruva-suryadigital committed:
Create custom Widget, formatted code

Create a custom Widget for list items.
Implement suggested changes.
Remove unnecessary comment.
Format code properly.

Amulya-Joshi commented on lib/main.dart:
```diff
@@ -1,115 +1,156 @@
-  // case the title) provided by the parent (in this case the App widget) and
-  // used by the build method of the State. Fields in a Widget subclass are
-  // always marked "final".
+class _MyAppState extends State<MyApp> {
```
Please add this in a different file.

Amulya-Joshi commented on lib/widgets.dart:
```diff
@@ -0,0 +1,27 @@
+import 'package:flutter/material.dart';
+import 'color.dart' as colors;
+
+class CustomContainer extends StatelessWidget {
```
Please rename this to be more specific.
